assets/js/components/teams/TeamShow.jsx

```diff
@@ -130,38 +114,54 @@ class TeamShow extends Component {
           membership={this.state.editingMembership}
           updateMembership={updateMembership}
         />
-      </DashboardLayout>
+      </div>
     )
-  }
-}
 
-function mapStateToProps(state, ownProps) {
-  const currentTeamId = state.auth.currentTeamId
-  const team = state.entities.teams[currentTeamId]
+    const activityView = (
+      <Paper className={classes.paper}>
+        <header className={classes.header}>
+          <Typography variant="headline" component="h3">
+            Team History
+          </Typography>
+        </header>
 
-  let memberships = []
-  if (team !== undefined) {
-    memberships = Object
-      .values(state.entities.memberships)
-  }
+        <AuditTable />
+      </Paper>
+    )
 
-  let invitations = []
-  if (team !== undefined) {
-    invitations = Object
-      .values(state.entities.invitations)
-      .filter(invitation => invitation.pending)
+    const tabs = [{
+      label: "Access View",
+      content: accessView,
+      path: "/teams/access",
+    }, {
+      label: "Activity View",
+      content: activityView,
+      path: "/teams/activity",
+    }]
+
+    if (userCan("view", "auditTrails", null, this.props.user)) {
```

Comment:
For the tabs, I need userCan to be hooked up to the store and get updates, or else the component gets the original null value for the user role. I was trying to use the UserCan component to achieve this behavior but couldn't think of a clean way since I just need to return true or false from userCan. 

Let me know what you think

Reply:
what if we defined 2 components, `<UserCan ...>` and `<UserCannot ...>`? Then you could have one thing displayed in the `true` case and a different thing in the `false` case